resources/elasticache-memcacheclusters.go

```diff
@@ -85,22 +102,36 @@ func (i *ElasticacheCacheCluster) Properties() types.Properties {
 	properties.Set("ClusterID", i.clusterID)
 	properties.Set("Status", i.status)
 
-	for _, tag := range i.Tags {
-		properties.SetTag(tag.Key, tag.Value)
+	if i.Tags != nil {
+		for _, tag := range i.Tags {
+			properties.SetTag(tag.Key, tag.Value)
+		}
 	}
 
 	return properties
 }
 
 func (i *ElasticacheCacheCluster) Remove(_ context.Context) error {
```

Comment:
If would change this to the following:

if serverless return DeleteServerlessCacheInput

drop the else 

and then run the deleteCacheCluster

A bit cleaner this way.